### ISSUE TITLE
Assorted accessibility issues

### ISSUE THREAD

**Issue by jelbourn** [angular/material.angular.io#671]
These issues were forwarded to me from a user that relies on assistive technology and was trying to use the site

### 1. The color contrast of everything in the left menu is insufficient.
That is generating 50+ Axe violations.

* The gray text has a contrast ratio of 3.88:1 and needs to be 4.5.1
 https://webaim.org/resources/contrastchecker/?fcolor=797979&bcolor=F2F2F2

* The blue text is dark enough
https://webaim.org/resources/contrastchecker/?fcolor=3F51B5&bcolor=F2F2F2

### 2. In the left menu, the currently selected item is indicated
only by a color change, you can never use color alone to communicate
information.

* For sighted users add an underline, arrow or dot for the selected
item (it's fine to keep the color too).
* For screen reader users, add aria-current="true" or
aria-current="page", either is fine.

### 3. Missing "accessible name" on action buttons.
       The “view source” buttons have no screen reader label

* The “Edit in Stack Blitz” buttons have no screen reader label.
Use aria-label to add these, e.g. <button aria-label="view source"....
Ideally these labels should be unique, so if you have multiple
examples on the same page the aria-label should be "view example 1
source code" but that's best practice.

### 4. The “Select a theme” button has a tooltip, but it does not have an
accessible name – the tooltip info is the name, but has been put into
aria-describedby which maps it as a description, not a label. Just
change "aria-describedby" to "aria-labelledby" (yes, 2 ls) and the
issue is fixed.

### 5. (headings/landmarks)
I) There is no heading or landmark that marks the start of the unique
content on an example page, which makes it hard for a screen reader
user to locate it.
There is a visually hidden <span> with the title, but it needs
role="heading" and aria-level="2" to be marked as an h2 heading.
E.g. (from: https://material.angular.io/components/form-field/overview)
```
<component-overview class="ng-star-inserted"
<!-- added role="heading" and aria-level="2" to make this an h2 heading). -->

><span tabindex="-1" class="cdk-visually-hidden" role="heading" aria-level="2"> Overview for form-field
</span>
```

II) Most headings have a link that is announced before the heading
content and makes them extremely annoying to navigate with a screen
reader:
```
<h3 id="form-field-appearance-variants" class="docs-header-link">

          <span header-link="form-field-appearance-variants">
<header-link class="ng-star-inserted">
<!-- This link needs to be moved outside the heading, preferably after
though before is acceptable, it just can't be part of the heading tag, ,
also remove the title, keep the aria-label) -->
<a title="Link to this heading" aria-label="Link to this heading"
class="docs-markdown-a"
href="/components/form-field/overview#form-field-appearance-variants"
aria-describedby="form-field-appearance-variants"><mat-icon
aria-hidden="true" role="img" class="mat-icon notranslate
material-icons mat-icon-no-color">link</mat-icon></a></header-link></span>
          Form field appearance variants

        </h3>
```
III) The main region of the page (the element that contains the
content unique to this page) is not programmatically identified.
Find the element that contains the unique page content(between the
global header and global footer), if there isn't one, add an extra div
to be that element, add role="main" (and Bob Dodson is your uncle).

IV) When you use more than one `<nav>` elements on a page you need to
label each, either using aria-label="name" or aria-labelledby="id of
element with visible label".
This applies to all the nav elements for the categories in the left menu.
e.g.
`<nav aria-label="Form controls">`

### 6. Over zealous screen reader labeling of toggle controls.
```
<!-- The aria-label should just be "form controls" no section toggle,
the aria-expanded attribute tells a screen reader that it is a toggle
control -->
<button cdkaccordionitem="" expanded="true"
class="docs-nav-content-btn ng-tns-c514-25"
aria-label="Form Controls, section toggle" aria-controls="panel-forms"
aria-expanded="true"> Form Controls <mat-icon class="mat-icon
notranslate ng-tns-c514-25 material-icons mat-icon-no-color"
aria-hidden="true" role="img">keyboard_arrow_up</mat-icon>
</button>
```

**Comment by Splaktar** [comment]
@jelbourn would you be okay with a more accessible theme picker that looked like the following?
![Theme-picker-purple-green-dark](https://user-images.githubusercontent.com/3506071/68539819-3259b780-0357-11ea-9a0f-e080cb1abfdb.png)
![Theme-picker-indigo-pink-light](https://user-images.githubusercontent.com/3506071/68539820-3259b780-0357-11ea-9333-8d37574808fb.png)

**Comment by Splaktar** [comment]
Feedback from Jeremy:
> I would like to show the colors as part of the item

**Comment by Splaktar** [comment]
Status of fixing these issues (in a yet to be posted PR):
- [x] 1. The color contrast of everything in the left menu is insufficient.
- [x] 2. In the left menu, the currently selected item is indicated
- [x] 3. Missing "accessible name" on action buttons.
- [ ] 4. The “Select a theme” button has a tooltip, but it does not have an accessible name
- [x] 5. Headings/landmarks I
- [ ] 5. Headings/landmarks II
- [x] 5. Headings/landmarks III
- [x] 5. Headings/landmarks IV
- [x] 6. Over zealous screen reader labeling of toggle controls.

**Comment by Splaktar** [comment]
OK, the bulk of this a11y work is done. Issue II in Headings/landmarks is going to be a significant change that will need coordination with this repo and `angular/components`, so that will need to be broken out into a separate issue.

I've posted PR https://github.com/angular/material.angular.io/pull/697 and added some screenshots there for the new visual preview of the theme picker colors.
